pkg/controller/logstash/pod.go

```diff
@@ -107,23 +109,32 @@ func getDefaultContainerPorts() []corev1.ContainerPort {
 }
 
 // readinessProbe is the readiness probe for the Logstash container
-func readinessProbe(useTLS bool) corev1.Probe {
-	scheme := corev1.URISchemeHTTP
-	if useTLS {
-		scheme = corev1.URISchemeHTTPS
+func readinessProbe(logstash logstashv1alpha1.Logstash) corev1.Probe {
+	var scheme = corev1.URISchemeHTTP
+	var port = network.HTTPPort
+	for _, service := range logstash.Spec.Services {
+		if service.Name == "api" {
+			port = int(service.Service.Spec.Ports[0].Port)
+		}
 	}
-	return corev1.Probe{
+	probe := corev1.Probe{
 		FailureThreshold:    3,
 		InitialDelaySeconds: 30,
 		PeriodSeconds:       10,
 		SuccessThreshold:    1,
 		TimeoutSeconds:      5,
 		ProbeHandler: corev1.ProbeHandler{
 			HTTPGet: &corev1.HTTPGetAction{
-				Port:   intstr.FromInt(network.HTTPPort),
+				Port:   intstr.FromInt(port),
 				Path:   "/",
 				Scheme: scheme,
 			},
 		},
 	}
+	return probe
+}
+
+// GetLogstashContainer returns the Logstash container from the given podSpec.
```

Comment:
nit: this seems to be only used in unit tests, let's move it to `pkg/controller/logstash/pod_test.go`?